scrapers_next/hi/people.py

```diff
@@ -1,89 +1,128 @@
+            name_parts = [x.strip() for x in regular_member.groups()]
+
+        # Extracts party abbreviation from list
+        party = name_parts.pop()
+
+        last_name, first_name = name_parts[0], " ".join(name_parts[1:])
+
+        dist_text = item.xpath("div/a")[0].text_content().strip()
+        chamber, district = [x.strip() for x in dist_text.split("District")]
+
+        contact_info = item.xpath("div/address")[0]
+        split_lines = contact_info.text_content().split("\r\n")
+        contact_list = [x.strip() for x in split_lines if len(x.strip())]
+
+        state_addr_base = "415 S Beretania St, Honolulu, HI 96813"
+        capitol_addr = f"{contact_list[0]}, {state_addr_base}"
+
+        cap_phone, cap_fax = [x.split(":")[-1].strip() for x in contact_list]
+
+        # Website does not allow scraping of member emails, so a manual check
+        #   of all member emails confirmed accuracy of below solution.
+        email_user = last_name.lower()
+        for string in self.social_remove_regexes:
+            email_user = string.sub("", email_user)
+        email_start = {"House": "rep", "Senate": "sen"}
+        email = email_start[chamber] + email_user + "@capitol.hawaii.gov"
+
+        soc_handles = {}
+        soc_links = contact_info.getnext().xpath("a")
+        for link in soc_links:
+            href = link.get("href").lower()
+            dom, han = self.soc_domain_and_handle_regex.search(href).groups()
+            for string in self.social_remove_regexes:
+                han = string.sub("", han)
+            soc_handles[dom] = han
+
+        # Website provides bad youtube handle for Sen Mike Gabbard
+        if f"{first_name} {last_name}" == "Mike Gabbard":
+            soc_handles["youtube"] = "senmikegabbard"
 
         p = ScrapePerson(
-            name=data["first_name"] + " " + data["last_name"],
+            name=f"{first_name} {last_name}",
             state="hi",
-            chamber=chamber,
-            district=data["district"],
-            given_name=data["first_name"],
-            family_name=data["last_name"],
-            party=party,
-            email=data["email"],
+            chamber="lower" if chamber[0] == "H" else "upper",
+            district=district,
+            given_name=first_name,
+            family_name=last_name,
+            party="Democratic" if party == "D" else "Republican",
+            image=member_photo,
+            email=email,
         )
-        p.capitol_office.address = address
-        p.capitol_office.voice = data["voice"]
-        p.capitol_office.fax = data["fax"]
-        p.add_source(data["url"])
-        p.add_link(data["url"])
+
+        if title:
+            p.extras["title"] = title
+
+        p.capitol_office.address = capitol_addr
+        p.capitol_office.voice = cap_phone
+        p.capitol_office.fax = cap_fax
+
+        p.add_source(self.source.url)
+        p.add_link(member_page)
+
+        if soc_handles["facebook"]:
```

Comment:
The social logic needs more work, I don't think it's running how you're expecting it to.

Reply:
Resolved in [newest commit](https://github.com/openstates/openstates-scrapers/pull/4247/commits/399cb4b2a6dff5f1160c4b686f77ecebc49576e1)